### DISCUSSION TITLE
Adding Scroll To Every Screen On the Expo Router

### DISCUSSION THREAD

**Discussion by chandan-jal-evolution** [expo/expo#30357] [Archived: Questions]
Hey there 👋

I'm currently working on a project with `Expo Router`, I'm working on the cross platform functionalities which supports `Android`, `iOS` and `web`.

I have this requirement to add the scrollbar to every screen as we know the contents will get bigger and bigger. I have created this `_layout.web.tsx` which handle the overall layout of the `web`.  I've also introduced the ScrollView over there it is working fine. 

Anyhow, my issue with the `Mobile Devices` the layout not working as expected. Here is the glimpse of the `_layout.tsx`

```
import React from "react";
import { Stack } from "expo-router/stack";
import { SafeAreaView } from "react-native-safe-area-context";

import Header from "@/components/dashboard/header";

export default function DashboardLayout() {
  return (
    <SafeAreaView className="flex-1">
      <Stack
        screenOptions={{
          header: () => <Header />,
        }}
      >
        <Stack.Screen
          name="index"
          options={{
            title: "Dashboard",
          }}
        />
        <Stack.Screen
          name="screen1"
          options={{
            title: "Screen 1",
          }}
        />
      <Stack.Screen
          name="long-content"
          options={{
            title: "Long Content",
          }}
        />
      </Stack>
    </SafeAreaView>
  );
}
```

So how can i implement the scroll view on each page. As I mentioned right I've also a `_layout.web.tsx` , here it is 

```import React from "react";
import { ScrollView, View } from "react-native";
import { Slot } from "expo-router";

import Header from "@/components/dashboard/header";
import Sidebar from "@/components/dashboard/sidebar";

export default function DashboardLayoutWeb() {
  return (
    <View className="flex-1 box-border flex flex-row">
      <Sidebar />
      <View className="flex-1 flex-col box-border">
        <Header />

        {/*
         * <Slot /> will replace all the content from specific routes.
         * <ScrollView /> React Native does not support auto scroll bar, for that we use ScrollView
         * So, if any pages/routes get more content than we'll get the scroll view automatically
         */}
        <ScrollView
          className="flex-1 px-3 py-7 sm:px-12 sm:py-9"
          contentContainerStyle={{ flexGrow: 1 }}
        >
          <Slot />
        </ScrollView>
      </View>
    </View>
  );
}
```

So, NO PAGE HAVE THE SCROLL VIEW, I'm implementing this on the layout files.

Please help me out here !!!!

**Comment by dennisofficial** [comment]
You need to implement the <ScrollView /> in your mobile layouts as well. Unless I am misinterpreting your issue

**Reply by chandan-jal-evolution** [reply]
Hey @dennisofficial  👋 

I've also did that as below 

```
<SafeAreaView className="flex-1">
  <ScrollView 
     className="flex-1" 
     contentContainerStyle={{
         flexGrow: 1,
     }}>
      <Stack
        screenOptions={{
          header: () => <Header />,
        }}
      >
      <Stack.Screen
        name="index"
        options={{
          title: "Dashboard",
        }}
      />
      <Stack.Screen
        name="screen1"
        options={{
          title: "Screen 1",
        }}
      />
    <Stack.Screen
        name="long-content"
        options={{
          title: "Long Content",
        }}
      />
    </Stack>
  </ScrollView>
</SafeAreaView>
```
 
but it is also causing some weird layout issue.

**Reply by dennisofficial** [reply]
There's no such thing as class names in mobile. Unless you are using some transformer. Use the style prop. Also try setting flex: 1 or flex grow: 1 to the parent. I get the same issue sometimes.

**Comment by dennisofficial** [comment]
Wait. I just realized. You set the scroll view in your layout where your stack is. You need the scroll view in the layout or the file of where your content is.

**Reply by dennisofficial** [reply]
Pretty much wrapping stack.screen doesn't work well.

**Reply by chandan-jal-evolution** [reply]
Yeah, so I want to wrap the `ScrollView` on the `_layout.tsx`

**Reply by dennisofficial** [reply]
You can utilize page groups to add an additional layout. That's your only best bet to wrap scroll view at the layout without wrapping the screen.
